dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumEventExt.kt

```diff
@@ -100,3 +101,71 @@ internal fun RumActionType.toSchemaType(): ActionEvent.Type1 {
         RumActionType.CUSTOM -> ActionEvent.Type1.CUSTOM
     }
 }
+
+internal fun NetworkInfo.toResourceConnectivity(): ResourceEvent.Connectivity {
+    val status = if (isConnected()) {
+        ResourceEvent.Status.CONNECTED
+    } else {
+        ResourceEvent.Status.NOT_CONNECTED
+    }
+    val interfaces = when (connectivity) {
+        NetworkInfo.Connectivity.NETWORK_ETHERNET -> listOf(ResourceEvent.Interface.ETHERNET)
+        NetworkInfo.Connectivity.NETWORK_WIFI -> listOf(ResourceEvent.Interface.WIFI)
+        NetworkInfo.Connectivity.NETWORK_WIMAX -> listOf(ResourceEvent.Interface.WIMAX)
+        NetworkInfo.Connectivity.NETWORK_BLUETOOTH -> listOf(ResourceEvent.Interface.BLUETOOTH)
+        NetworkInfo.Connectivity.NETWORK_2G,
+        NetworkInfo.Connectivity.NETWORK_3G,
+        NetworkInfo.Connectivity.NETWORK_4G,
+        NetworkInfo.Connectivity.NETWORK_5G,
+        NetworkInfo.Connectivity.NETWORK_MOBILE_OTHER,
+        NetworkInfo.Connectivity.NETWORK_CELLULAR -> listOf(ResourceEvent.Interface.CELLULAR)
+        NetworkInfo.Connectivity.NETWORK_OTHER -> listOf(ResourceEvent.Interface.OTHER)
+        NetworkInfo.Connectivity.NETWORK_NOT_CONNECTED -> emptyList()
+    }
+
+    val cellular = if (cellularTechnology != null || carrierName != null) {
+            ResourceEvent.Cellular(
+                technology = cellularTechnology,
+                carrierName = carrierName
+            )
+        } else {
+            null
+        }
+    return ResourceEvent.Connectivity(
+        status, interfaces, cellular
+    )
+}
+
+internal fun NetworkInfo.toErrorConnectivity(): ErrorEvent.Connectivity {
```

Comment:
What about the ViewEvent and the ActionEvent ?

Reply:
In theory, only resources and errors should have network info